compiler/luci/service/src/Nodes/CircleFullyConnected.cpp

```diff
@@ -36,4 +40,96 @@ luci::CircleNode *CloneNodeLet<CN::DEF>::visit(const luci::CircleFullyConnected
   return cloned;
 }
 
+namespace sinf
+{
+
+loco::TensorShape Algorithm::visit(const luci::CircleFullyConnected *node)
+{
+  auto input_shape = circle_shape(loco::must_cast<CircleNode *>(node->input()));
+  auto weights_shape = circle_shape(loco::must_cast<CircleNode *>(node->weights()));
+
+  loco::TensorShape out_shape;
+
+  // NOTE Some recipes in some repositories are using rank 4 input for FullyConnected.
+  //      Until they are all fixed, disable following assert.
+  // TODO Enable following assert after related fixes are applied
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L194
+  // LUCI_ASSERT(input_shape.rank() == 2 || input_shape.rank() == 3,
+  //             "Input rank of FullyConnected should be 2 or 3");
+
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L225
+  LUCI_ASSERT(weights_shape.rank() == 2, "Weights of FullyConnected should be 2");
+  LUCI_ASSERT(weights_shape.dim(0).known() && weights_shape.dim(1).known(),
+              "Weights of FullyConnected should be known")
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L353-L367
+
+  /*
+   * [CircleNode]       [CircleNode]
+   *   (input)            (weight)
+   *         \            /
+   *     [CircleFullyConnected]
+   *               |
+   *            (output)
+   *
```

Comment:
```suggestion
```

(optional suggestion) 

I think we could remove this graph. 
I guess the reader can easily catch the meaning of `input_shape` and `weight_shape` by referencing lines 48-49.

Reply:
It's fine with me. I agree with your opinion, too.
I will delete it, but wouldn't it be clearer if it's displayed as a graph?

Reply:
> but wouldn't it be clearer if it's displayed as a graph?

Yes, I agree with your point somehow :) 

If you want to revive the graph, you could do it! I'm not opposed to do.